### DISCUSSION TITLE
How would you test against policies which are stored in a database?

### DISCUSSION THREAD

**Discussion by bbodensieck** [cerbos/cerbos#1589] [Q&A] [answered]
I really like your way of testing like described [here](https://docs.cerbos.dev/cerbos/latest/policies/compile.html#testing). Currently I am thinking about which way is best to store policies. Because I would like to use the API for changes while runtime I would like to use MySQL. But I am struggling with testing.

How would you test against policies which are stored in a database?

Thanks again,
Björn

**Comment by charithe** [comment, marked as answer]
The Cerbos testing framework is designed with the GitOps workflow in mind. The expectation is that any policy changes would go through a review process before being promoted to production. During that process the tests can be automatically run by the CI system to catch any problems. 

If you are using a database to store the policies because they need to modified on the fly, then does it make much sense to test them because you'd be doing that after they have already gone live?

If you still want to go ahead with testing the policies you have in the database, you could use the [`cerbosctl get` command](https://docs.cerbos.dev/cerbos/latest/cli/cerbosctl.html#get) to pull down the currently active policies and write them to disk. Then you can copy over your test files to the same location and use the `cerbos compile` command to run the tests.

**Reply by bbodensieck** [reply]
Makes sense. Thank you
